cns/restserver/internalapi.go

```diff
@@ -195,28 +196,43 @@ func (service *HTTPRestService) syncHostNCVersion(ctx context.Context, channelMo
 	for _, nc := range ncList.Containers {
 		newHostNCVersionList[nc.NetworkContainerID] = nc.Version
 	}
-	for _, ncID := range hostVersionNeedsUpdateContainers {
+	for ncID := range hostVersionNeedsUpdateContainers {
 		versionStr, ok := newHostNCVersionList[ncID]
 		if !ok {
+			// NMA doesn't have this NC that we need programmed yet, bail out
 			continue
 		}
 		version, err := strconv.Atoi(versionStr)
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse container version of %s", ncID)
 		}
-
 		// Check whether it exist in service state and get the related nc info
 		ncInfo, exist := service.state.ContainerStatus[ncID]
 		if !exist {
 			return errors.Wrapf(errNonExistentContainerStatus, "can't find NC with ID %s in service state, stop updating this host NC version", ncID)
 		}
+		hostVersion, err := strconv.Atoi(ncInfo.HostVersion)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse host nc version string %s", ncInfo.HostVersion)
+		}
+		if hostVersion > version {
```

Comment:
Can we call it newHostVersion? version is too generic

Reply:
refactored all the variable names here to make it very clear which is which.